workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogHolder.kt

```diff
@@ -93,6 +131,7 @@ internal class DialogHolder<T : Overlay>(
     @Suppress("UNCHECKED_CAST")
     this.rendering = rendering as T
     this.environment = environment
+    allowEvents = !modal || !environment[CoveredByModal]
```

Comment:
This is another aside - but since we take such care in locking up this modal functionality (and in some cases making it tighter to avoid certain edge cases) - why are we using `Dialog` types at all? Just for the separate window? Couldn't we just create our own `Window`? I guess i get it for `AlertDialog` since that gives us some simpled themed UI.

Reply:
I don't think this is a `Dialog` issue, I think it's a `Window` issue. And `Window` is a PITA to use directly. Also remember that we're allowing app developers to build their own. Don't want to force them to use `Window` for that.